lib/stat_tracker.rb

```diff
@@ -87,34 +87,51 @@ def count_of_teams
        @teams_stats_data.size
     end
    
-    def best_offense
-
-        teams_goals = Hash.new(0)
-        games_played = Hash.new(0)
+    def team_seasons_goals
+        teams_data = Hash.new { |hash, key| hash[key] = { goals: 0, games_played: 0 } }
 
         @seasons_stats_data.each do |game_id, season|
-            teams_goals[season.team_id] += season.goals
-            games_played[season.team_id] += 1
+            teams_data[season.team_id][:goals] += season.goals
```

Comment:
Great refactor, much cleaner to make a nested data collection instead of 2 separate hashes.